lib/spack/spack/database.py

```diff
@@ -319,7 +324,10 @@ def _check_ref_counts(self):
         counts = {}
         for key, rec in self._data.items():
             counts.setdefault(key, 0)
-            for dep in rec.spec.dependencies.values():
+            # XXX(deptype): This checks all dependencies, but build
+            #               dependencies might be able to be dropped in the
+            #               future.
```

Comment:
Is this consistent with the ref counting?  It looks like you only increment for link here, but this checks for all.